prince-of-versions/src/main/java/co/infinum/princeofversions/mvp/interactor/impl/PovInteractorImpl.java

```diff
@@ -25,22 +24,38 @@ public PovInteractorImpl(VersionVerifier versionVerifier, UpdateConfigLoader loa
     @Override
     public void checkForUpdates(final PovInteractorListener listener) {
         versionVerifier.verify(loader, new VersionVerifierListener() {
-            @Override
-            public void versionAvailable(VersionContext version) {
-                if (version.isCurrentLessThanMinimum()) {
-                    listener.onMandatoryUpdateAvailable(version);
-                } else if (version.hasOptionalUpdate() && version.isCurrentLessThanOptional()) {
-                    listener.onUpdateAvailable(version);
-                } else {
-                    listener.onNoUpdateAvailable(version);
-                }
-            }
+                    @Override
+                    public void versionAvailable(VersionContext version) {
```

Comment:
I think this method should work slightly different.
 * If `version.isCurrentLessThanMinimum` is `true` then there is mandatory update only if `minimum_version_min_sdk` less than or equal to device's sdk number. Nothing to check with optional update here.
 * If there is no mandatory update then we should check optional update if optional update is present. There is optional update only if `version.isCurrentLessThanOptional` is true and latest version's `min_sdk` is less than or equal to device's sdk number.

Please @reisub and @ikocijan confirm or deny this.